### PR TITLE
fix: Implements connection pool for postgres online store

### PR DESCRIPTION
**Pull request by Natan16**
**What this PR does / why we need it**:
When using a postgres online store, there's a chance the single connection times-out due to inactivity, so we get: 
    InterfaceError
    connection already closed 
The mainstream solution for that is implementing a connection pool with psycopg in order to guanratee the connections to the database are managed correctly.

**Which issue(s) this PR fixes**:
https://github.com/feast-dev/feast/issues/3611

Fixes #

### CODE REVIEW

File: sdk/python/feast/infra/online_stores/contrib/postgres.py [hunk 1]
```diff
@@ -13,26 +14,28 @@
 from feast.feature_view import FeatureView
 from feast.infra.key_encoding_utils import serialize_entity_key
 from feast.infra.online_stores.online_store import OnlineStore
-from feast.infra.utils.postgres.connection_utils import _get_conn
+from feast.infra.utils.postgres.connection_utils import _get_connection_pool
```
Comment (avelino) [inline sdk/python/feast/infra/online_stores/contrib/postgres.py:17]: We are forcing `pool` connection **by default**, we should leave it configurable by the user (he should choose the connection time that will be used), force `pool` connection in small project will be *"expensive"* (unnecessary resource usage).

**My suggestion** would be to let the user **choose** the connection type, **not `pool`** by *default* and the projects that need pool define it as pool

File: sdk/python/feast/infra/utils/postgres/postgres_config.py [hunk 2]
```diff
@@ -6,6 +6,8 @@
 
 
 class PostgreSQLConfig(FeastConfigBaseModel):
+    min_conn: int = 1
+    max_conn: int = 10
```
Comment (avelino) [inline sdk/python/feast/infra/utils/postgres/postgres_config.py:10]: create a connection definition variable that will be used from `_get_conn()`

**e.g:**
```py
type_conn: str = "normal" # or "pool"
```

**Review by avelino — CHANGES_REQUESTED** [review]

**Comment by avelino** [comment]
> Ok, I added back the possibility to use a single connection, now with the parameter `keepalives_idle` in case the database is configured to drop idle connections. 0 uses the system default.

@Natan16 great, I think it is important to write a unit test to validate the behavior that was implemented

@adchia are we going in the best direction for the project?

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**